0013-delete-enqueued-updates.md

```diff
@@ -0,0 +1,151 @@
+- Title: Delete enqueued updates
+- Start Date:
+- Specification PR: https://github.com/meilisearch/specifications/pull/13
+- MeiliSearch Issue: 
+    - https://github.com/meilisearch/MeiliSearch/issues/1104
+- Parent Spec: https://github.com/meilisearch/specifications/pull/12
+
+
+# Delete enqueued updates
+
+## 1. Feature Description and Interaction
+
+### Summary
+
+Allow to delete messages in the update queue or to empty the entire queue. A deleted update will have the status `aborted`.
+
+Questions not answered:
+- Should we delete in-process updates? If yes, what the default behavior for the clear method. What are the side effects?
+- We have an attribute on update info `processedAt`. What becomes this attribute if we add the possibility to abort an update? We already use this term in the case of a failed update.
+
+### Motivation
+
+The update queue is convenient because it allows us to serialize updates and assure the [ACID property](https://en.wikipedia.org/wiki/ACID) of MeiliSearch. It also causes some problems. As the queue message is an append-only queue, it is impossible to remove an update from this queue. The updates can be large and can take a long time to index. Sometimes, it would be convenient to delete a message in this queue or empty it. 
+
+### Additional Materials
+
+As explained on the [parent spec](https://github.com/meilisearch/specifications/pull/12), It's quite rare to have this kind of update queue. Elasticsearch or Algolia, who have this king of update queue, don't give the delete or clear their 'tasks' queues. 
+
+### Explanation
+
+To make this feature possible, two new routes and a new status (`aborted`) must be added. 
+
+An aborted update will stay at its place on the queue. So the typical answer of the get all update status route will be: 
+
+```json
+[
+    {
+        "status": "processed",
+        "updateId": 1,
+        ...
+    },
+    {
+        "status": "failed",
+        "updateId": 2,
+        ...
+    },
+    {
+        "status": "processed",
+        "updateId": 3,
+        ...
+    },
+    {
+        "status": "enqueued",
+        "updateId": 4,
+        ...
+    },
+    {
+        "status": "aborted",
+        "updateId": 5,
+        ...
+    },
+    {
+        "status": "enqueued",
+        "updateId": 6,
+        ...
+    },
+    ...
+]
+```
+
+the update status document will look like:
+
+```json
+{
+  "status": "aborted",
+  "updateId": 12,
+  "type": {
+    "name": "DocumentsAddition",
+    "number": 4
+  },
+  "duration": 0,
+  "enqueuedAt": "2019-12-07T21:16:09.623944Z",
+  "processedAt": "2019-12-07T21:16:09.703509Z" // Date of deletion
+}
+```
+
+
+
+#### HTTP API
+
+**Delete an update:**
+
+- Method: DELETE
+- Route: `/indexes/:index_uid/updates/:update_id`
+    - index_uid: The index uid
+    - update_id: The update identifier
+- Status Code: 204
+- Response: No Body
+
+Other proposition:
+
+- Status Code: 200
+- Response: 
+```json
+{
+    deletedUpdate: 12
+}
+```
```

Comment:
If I understand well, this action is not asynchronous, like an index deletion. So I would rather stay consistent with the `index` routes behavior, which is `204` and no body answer.

Reply:
(Sorry if it was not the step of the process to give my opinion. Anyway, you can add on the spec "the `204` solution" is the consistent solution with `indexes` routes 😇)

Reply:
I proposed two solutions. The first one is like you said a `204` with no content. And I think it's the best choice for this route. But the second proposition is made to be consistent with the clear all updates proposition. See the comment on your second comment.

Reply:
204 No Content for everything! 👍